packages/hardware-ledger/src/LedgerKeyAgent.ts

```diff
@@ -377,12 +377,18 @@ export class LedgerKeyAgent extends KeyAgentBase {
   }
 
   private static attachDisconnectionCleanupHandler(transport: LedgerTransportType) {
-    const onDisconnect = () => {
+    const onDisconnect = async () => {
       transport.off('disconnect', onDisconnect);
       this.deviceConnections = this.deviceConnections.filter(
         ({ deviceConnection }) => deviceConnection.transport !== transport
       );
-      void transport.close();
+
+      try {
+        await transport.close();
+      } catch {
```

Comment:
Can the error be inspected to ensure it's the expected failure before ignoring?

Reply:
addressed